src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/Buffering/HttpRequestBufferLoggerBuilderExtensions.cs

```diff
@@ -0,0 +1,97 @@
+
+namespace Microsoft.Extensions.Logging;
+
+/// <summary>
+/// Lets you register log buffers in a dependency injection container.
+/// </summary>
+[Experimental(diagnosticId: DiagnosticIds.Experiments.Telemetry, UrlFormat = DiagnosticIds.UrlFormat)]
+public static class HttpRequestBufferLoggerBuilderExtensions
+{
+    /// <summary>
+    /// Adds HTTP request-aware buffer to the logging infrastructure. Matched logs will be buffered in
+    /// a buffer specific to each HTTP request and can optionally be flushed and emitted during the request lifetime./>.
+    /// </summary>
+    /// <param name="builder">The <see cref="ILoggingBuilder" />.</param>
+    /// <param name="configuration">The <see cref="IConfiguration" /> to add.</param>
+    /// <returns>The value of <paramref name="builder"/>.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="builder"/> is <see langword="null"/>.</exception>
+    public static ILoggingBuilder AddHttpRequestBuffer(this ILoggingBuilder builder, IConfiguration configuration)
+    {
+        _ = Throw.IfNull(builder);
+        _ = Throw.IfNull(configuration);
+
+        return builder
+            .AddHttpRequestBufferConfiguration(configuration)
+            .AddHttpRequestBufferProvider();
+    }
+
+    /// <summary>
+    /// Adds HTTP request-aware buffer to the logging infrastructure. Matched logs will be buffered in
+    /// a buffer specific to each HTTP request and can optionally be flushed and emitted during the request lifetime./>.
+    /// </summary>
+    /// <param name="builder">The <see cref="ILoggingBuilder" />.</param>
+    /// <param name="level">The log level (and below) to apply the buffer to.</param>
+    /// <param name="configure">The buffer configuration options.</param>
+    /// <returns>The value of <paramref name="builder"/>.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="builder"/> is <see langword="null"/>.</exception>
+    public static ILoggingBuilder AddHttpRequestBuffer(this ILoggingBuilder builder, LogLevel? level = null, Action<HttpRequestBufferOptions>? configure = null)
+    {
+        _ = Throw.IfNull(builder);
+
+        _ = builder.Services
+            .Configure<HttpRequestBufferOptions>(options => options.Rules.Add(new BufferFilterRule(null, level, null)))
+            .Configure(configure ?? new Action<HttpRequestBufferOptions>(_ => { }));
+
+        return builder.AddHttpRequestBufferProvider();
+    }
+
+    /// <summary>
+    /// Adds HTTP request buffer provider to the logging infrastructure.
+    /// </summary>
+    /// <param name="builder">The <see cref="ILoggingBuilder" />.</param>
+    /// <returns>The <see cref="ILoggingBuilder"/> so that additional calls can be chained.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="builder"/> is <see langword="null"/>.</exception>
+    public static ILoggingBuilder AddHttpRequestBufferProvider(this ILoggingBuilder builder)
```

Comment:
Does this need to be public? I'm not sure why external code would want to call this directly.